CRM/MembershipExtras/Hook/ValidateForm/MembershipUpdate.php

```diff
@@ -0,0 +1,156 @@
+
+/**
+ * Class CRM_MembershipExtras_Hook_ValidateForm_MembershipUpdate.
+ *
+ * Implements hook to validate the membership update form.
+ */
+class CRM_MembershipExtras_Hook_ValidateForm_MembershipUpdate {
+
+  /**
+   * Form object that is being validated.
+   *
+   * @var \CRM_Member_Form_Membership
+   */
+  private $form;
+
+  /**
+   * List of the submitted fields and their values passed from the hook.
+   *
+   * @var array
+   */
+  private $fields;
+
+  /**
+   * List of form validation errors passed from the hook.
+   *
+   * @var array
+   */
+  private $errors;
+
+  /**
+   * CRM_MembershipExtras_Hook_BuildForm_Membership constructor.
+   *
+   * @param \CRM_Member_Form $form
+   */
+  public function __construct(CRM_Member_Form &$form, &$fields, &$errors) {
+    $this->form = $form;
+    $this->fields = &$fields;
+    $this->errors = &$errors;
+  }
+
+  /**
+   * Validates the form.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function validate() {
+    if ($this->isPaymentPlan() && $this->isMembershipTypeChanged()) {
+      $this->errors['membership_type_id'] = ts(
+        'This membership is part of an ongoing payment plan and cannot be 
+        edited directly. To modify the active memberships in this payment plan 
+        please go to the manage future instalments screen which can be found as 
+        an action on the most recent recurring contribution record. There you 
+        can add or remove memberships.'
+      );
+    }
+  }
+
+  /**
+   * Checks if the membership was paid for with a payment plan.
+   *
+   * @return bool
+   *   TRUE if membership is part of a payment plan, FALSE otherwise.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function isPaymentPlan() {
+    $contribution = $this->getLastMembershipPayment();
+
+    if (empty($contribution['contribution_recur_id'])) {
+      return FALSE;
+    }
+
+    $recurringContribution = $this->getRecurringContribution($contribution['contribution_recur_id']);
+    $processorID = CRM_Utils_Array::value('payment_processor_id', $recurringContribution);
+    $isManualPaymentPlan = ManualPaymentProcessors::isManualPaymentProcessor($processorID);
+
+    if ($isManualPaymentPlan) {
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Obtains payment processor ID used for the last membership payment.
+   *
+   * Returns tha las payment processor ID used to pay for the membership with a
+   * recurring contribution.
+   *
+   * @return mixed
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function getLastMembershipPayment() {
+    $contribution = civicrm_api3('MembershipPayment', 'get', [
+      'sequential' => 1,
+      'membership_id' => $this->form->_id,
+      'options' => [
+        'limit' => 1,
+        'sort' => 'contribution_id DESC',
+      ],
+      'return' => [
+        'contribution_id',
+        'contribution_id.contribution_recur_id',
+      ],
+    ]);
+
+    return $contribution['values'][0];
```

Comment:
this will throw a notice if no record found, probably you need to add an if statement and return NULL if no result found